Filters navbar bioclimatic variables

aabdaab opened the pull request:
This PR adds a navbar with dropdowns to `Bioclimatic Vars` page.
* It's pulling the data from GraphCMS
* Persist the state in the url
* Url as a source of truth
* Added `url.js` helper file to manage url query params
* Avoid setting `from` years that are subsequent to `to` years and the other way around (by passing `enabledYears` prop to dropdown)
* Download button doesn't do anything for now
<img width="1436" alt="Screenshot 2019-10-22 at 11 37 35" src="https://user-images.githubusercontent.com/6136899/67278186-64969880-f4c0-11e9-82b3-11e646cda4f4.png">

**Note:**
Check for Tanzania as it's the one that has data in the CMS.

aabdaab committed:
Add .env.sample file and include .env in gitignore

aabdaab committed:
Add dropdowns navbar to bioclimatic vars page, preserve the state in the url

kfilip94 commented:
:eyes: checking

aabdaab committed:
Derive years from the url state

kfilip94 commented:
- Remove border-left from the filters bar:
![image](https://user-images.githubusercontent.com/15097138/67302161-5fe5da80-f4e8-11e9-8289-a828421e13fe.png)
- I'd increase the width of the filter dropdown closer to design value (144px),
- On the design I see `From current` value, should it be added to the first dropdown?
![image](https://user-images.githubusercontent.com/15097138/67304995-97ef1c80-f4ec-11e9-95a4-4c7a82ae5c18.png)
- Maybe it's a corner case but if you set the query params in the wrong way in the URL, (eg. fromYear = 2070 and toYear = 2050), these params are present in the dropdown:
![image](https://user-images.githubusercontent.com/15097138/67304289-83f6eb00-f4eb-11e9-9b40-c992bdd4679b.png)
- I think you can change the hover color of the dropdown element to `$color-background` (#EDECEA) because according to the design, the current one is too dark.
![image](https://user-images.githubusercontent.com/15097138/67305923-0bddf480-f4ee-11e9-80d5-d302c593b631.png)

I'm checking the code now :mag:

aabdaab committed:
Remove border-left on the first dropdown

kfilip94 commented on src/components/dropdown/styles.scss:
```diff
@@ -26,11 +29,35 @@
+  }
+}
+
+.dd-list {
```
Instead of using `& > *:last-child > *:last-child ` I'd suggest to set border on the `ul` element (list), and `border-bottom` to the `li` elements (list items)

```suggestion
.dd-list {
  border: 1px solid $color-gray-1;
  & > *:not(:last-child) {
    border-bottom: 1px solid $color-gray-1;
  }
}
```

aabdaab replied:
👍

aabdaab committed:
Fix titles of year dropdowns, increase min-width of dropdown

kfilip94 commented on src/components/dropdown/styles.scss:
```diff
@@ -26,11 +29,35 @@
+      height: 44px;
+      cursor: pointer;
+      background-color: $color-white;
+      border: {
```
Check my comment below, you can remove the `border` property from here and apply it to the list elements below
```suggestion
```

kfilip94 commented on src/components/filters/index.js:
```diff
@@ -1,3 +1,117 @@
+  const setToYear = year =>
+    setQueryParams({ ...currentQueryParams, toYear: year }, location, history);
+  const setQueryParamsYears = (fromYear, toY) =>
+    setQueryParams({ ...currentQueryParams, fromY: fromYear, toYear: toY }, location, history);
```
Eslint is complaining about the order of those functions:
![image](https://user-images.githubusercontent.com/15097138/67309631-1dc29600-f4f4-11e9-8c3b-185b928e7f63.png)

aabdaab replied:
👍 fixed

aabdaab commented:
Regarding the issue with directly changing sth in the url and getting wrong results, I think it's a wider issue and we should tackle it in a different PR; now if you put localhost:3000/bleble you get an error as well.

aabdaab committed:
Reorder index.js file

aabdaab committed:
Improve css

aabdaab commented:
Ready again ☝️

kfilip94 commented:
I found a bug :(
When you click again on already selected option in the third dropdown (As usual, Optimistic, Current), this happens:
![ogh5o-2qfl3](https://user-images.githubusercontent.com/15097138/67310703-3469ec80-f4f6-11e9-8d07-dbb4bf988d5e.gif)

Also, there is a quick 'undefined' text flash when changing the selected option

aabdaab committed:
Fix url state setting

aabdaab committed:
Fetch years together with scenarios

aabdaab commented:
☝️ Ready again. I changed the fetching of years, as there was some lag due to fetching separately scenarios and years for a chosen scenario. Now, there's only one call to the API and you get the relevant years with that scenario - that avoids calling couple of use effects, now we only need one for setting a default url state once the scenarios are fetched. That also fixed a bug with selecting already selected option.

aabdaab committed:
Reset enabled years to all available ones after changing scenario

kfilip94 reviewed (approved):
@aabdaab Thank you for your changes, excellent work! :1st_place_medal:
